### ISSUE TITLE
Linear regression learnt parameters are wrong when using sklearn with my specific docker instance. 

### ISSUE THREAD

**Issue by raghul-parthipan** [tensorflow/tensorflow#52814]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Custom code
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): Binary? Installed via docker pull tensorflow/tensorflow:2.4.0-jupyter
- TensorFlow version (use command below): 2.4.0
- Python version: 3.6.9
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: CUDA 11.0
- GPU model and memory: NVIDIA GeForce GTX 1650 Ti/PCIe/SSE2

**Describe the current behavior**

The docker tf container is causing an issue with sklearn's linear regression function (and I'm not sure what else it is causing an issue with). 

After running a tf docker container which was pulled from one of the tf images, I seek to fit a polynomial model using sklearn. 

I find that the coefficients which are given are not the coefficients which are found if the regression problem is explicitly solved for using maximum likelihood.

I have checked and I see that when everything is run with a non-tf docker (e.g. jupyter one) or just using a conda env, the issue is no longer there.

Further checks included running the same tf docker container on a mac (instead of my ubuntu 20.04). On doing so there was no issue. 

I have also tried running the same tf docker in a virtual machine ubuntu 20.04 to see if my machine was causing the bug. But the issue persists in the virtual machine ubuntu 20.04.

Moreover, I find that when the number of data points is fewer, there is no issue. Only when the number of data points is increased is this issue there.

Hence I don't know if it is particular to this combination of ubuntu + the tf docker + use of sklearn. Or just something else.

I've included alternative ways to get at the coefficients (by training a tf model with mse loss --- this gives the correct answer), and using the statsmodels library (which gives the wrong answer), for reference.

**Describe the expected behavior**

Expected behaviour - sklearn's linear regression coefficients should match with those derived by solving the maximum likelihood estimation set-up. 

- Do you want to contribute a PR? (yes/no): No
- Briefly describe your candidate solution(if contributing): N/A

**Standalone code to reproduce the issue**

To reproduce:

1. Need to create an ubuntu 20.04 environment (e.g. through virtual box). Although you can proceed without doing this, but i don't know if the issue will then be reproduced.
2. Download the following data (https://drive.google.com/file/d/1NkQJbgn2cG-4y30AcEn1olthpROFWrLo/view?usp=sharing) and notebook  (https://drive.google.com/file/d/1XKEGgwoQ-3OyQYie9-HNI4w0mqNMPWo1/view?usp=sharing) into your working directory. 
3. You need to have docker installed and configured 
4. From the working directory run the following:
5. `docker run --name abc --user "$(id -u):$(id -g)" -v "$(pwd)":/tf -it -p 8888:8888 tensorflow/tensorflow:2.4.0-jupyter`
6. Kill this with "ctrl+c"
7. `docker start abc`
8.  `docker exec -it abc pip install --no-cache-dir --upgrade pip`
9. `docker start -i abc`
10. Now you should be able to run the notebook and see the issue.

**Comment by mohantym** [comment]
Hi @Saduf2019! Could you please look at this issue?

**Comment by sachinprasadhs** [comment]
Hi, Have you tried the same code outside docker? Just to understand the root cause of your problem.

**Comment by raghul-parthipan** [comment]
Hi @sachinprasadhs yes I have - I wrote in the initial comments under "Describe the current behavior" that in a non-tf docker it works and in a separate conda env (no docker) with tf it works too.

**Comment by sachinprasadhs** [comment]
Then in that case it doesn't looks like TF issue, did you check the same issue in the docker repo.

**Comment by raghul-parthipan** [comment]
But, as written above, when I run it in a non-TF docker then there is no issue. So it looks like it is a result of ubuntu 20.04 + docker + tf. Hence I made the issue here as with the non-TF docker there isn't an issue.
